frontend/app-development/features/textEditor/TextEditor.tsx

```diff
@@ -21,7 +21,17 @@ const storageGroupName = 'textEditorStorage';
 
 export const TextEditor = () => {
   const [searchParams, setSearchParams] = useSearchParams({ lang: '', search: '' });
-  const selectedLangCodes = searchParams.get('lang').split('-');
+  const [selectedLangCodes, setSelectedLangCodes] = useState<LangCode[]>([]);
+
+useEffect(() => {
+  const initialSelectedLangCodes = JSON.parse(localStorage.getItem('selectedLanguages')) || [];
+  setSelectedLangCodes(initialSelectedLangCodes);
+}, []);
```

Comment:
Here is a suggestion to set the default language when the selectedLanguages array is empty:

```suggestion
  const handleSelectedLangCodes = (value: LangCode[]) => {
    setSelectedLangCodes(value?.length > 0 ? value : [defaultLangCode]);
  };

  useEffect(() => {
    const initialSelectedLangCodes = JSON.parse(localStorage.getItem('selectedLanguages'));
    handleSelectedLangCodes(initialSelectedLangCodes);
  }, []);
```

If you use this solution, you probably also need to replace `setSelectedLangCodes={setSelectedLangCodes}` with `setSelectedLangCodes={handleSelectedLangCodes}`

Reply:
I agree 👍 , done